src/main/java/io/github/opencubicchunks/cubicchunks/cubicgen/common/gui/component/UIBlockStateButton.java

```diff
@@ -53,11 +56,11 @@
 import net.minecraft.tileentity.TileEntity;
 import net.minecraft.util.math.BlockPos;
 
-public class UIBlockStateButton<T extends UIBlockStateButton<T>> extends UIComponent<T> {
+public class UIBlockStateButton extends UIComponent<UIBlockStateButton> {
```

Comment:
The reason for it being done this way is that you can still extend the class and get the correct return types from builder-style methods.